clip.cpp

```diff
@@ -1168,13 +1216,91 @@ static void feature_out(std::vector<tile_feature> const &features, mvt_layer &ou
 			// features that will not
 
 			for (size_t i = 1; i < features.size(); i++) {
+				std::set<std::string> keys;
+
+				for (size_t j = 0; j + 1 < features[i].tags.size(); j += 2) {
+					const std::string &key = features[i].layer->keys[features[i].tags[j]];
+					keys.insert(key);
+				}
+
 				for (size_t j = 0; j + 1 < features[i].tags.size(); j += 2) {
-					std::string key = features[i].layer->keys[features[i].tags[j]];
+					const std::string &key = features[i].layer->keys[features[i].tags[j]];
 
 					auto f = attribute_accum.find(key);
 					if (f != attribute_accum.end()) {
 						serial_val val = mvt_value_to_serial_val(features[i].layer->values[features[i].tags[j + 1]]);
 						preserve_attribute(f->second, key, val, full_keys, full_values, attribute_accum_state);
+					} else if (accumulate_numeric.size() > 0) {
+						const mvt_value &val = features[i].layer->values[features[i].tags[j + 1]];
+						if (val.is_numeric()) {
+							// If this is a numeric attribute, but there is also a tippecanoe:sum (etc.) for the
+							// same attribute, we want to use that one instead of this one.
+
+							for (auto const &op : numeric_operations) {
+								std::string compound_key = accumulate_numeric + ":" + op.first + ":" + key;
+								auto compound_found = keys.find(compound_key);
+								if (compound_found != keys.end()) {
```

Comment:
Let's factor out into a separate function just to tamp down the nesting complexity